### PR TITLE
Decrease ORA auto-score from 14 to 3 days with optimisation

### PR DESCRIPTION
**Pull request by mabdullahz**
### Description

[LP-2303](https://philanthropyu.atlassian.net/browse/LP-2303)

This PR does not have the unit tests as was decided. We will create a separate PR for the refactored code's unit tests, this only includes the functionality for LP-2303 and LP-1678

Find unit test in PR https://github.com/philanthropy-u/edx-platform/pull/1754

### CODE REVIEW

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

File: common/djangoapps/student/tests/factories.py [hunk 1]
```diff
@@ -154,7 +154,14 @@ def _create(cls, model_class, *args, **kwargs):
...
 
-        return manager.create(*args, **kwargs)
+        course_enrollment = manager.create(*args, **kwargs)
+
+        if course_enrollment is not None and 'created' in kwargs:
+            # Let factory add created datetime
+            course_enrollment.created = kwargs['created']
+            course_enrollment.save()
+
+        return course_enrollment
```
Comment (muak50) [inline common/djangoapps/student/tests/factories.py:164]: This is a core change, lets comment it's needs and usage.

**Review by muak50 — CHANGES_REQUESTED** [review]
Please look into suggested changes.

File: common/djangoapps/student/tests/factories.py (hunk 1 above)
Comment (zahid-arbisoft) [reply common/djangoapps/student/tests/factories.py:164]: This change is not required in the unit test now, so reverting it back.

File: openedx/features/assessment/helpers.py [hunk 2]
```diff
@@ -1,103 +1,128 @@
...
+        if submission:
+            submissions.append(submission)
+
+    return submissions
+
+
+def _log_multiple_submissions_info(submissions_to_autoscore, days_to_wait, delta_date):
+    """
+    Log ORA auto scoring status
+    """
```
[the remark this answers is not in the document]
Comment (zahid-arbisoft) [reply openedx/features/assessment/helpers.py:100]: done

**Comment by zahid-arbisoft** [comment]
@zeshanarifios @muak50 Doc style in this PR is sphinx, I have updated it to google style in unit test PR

**Review by muak50 — APPROVED** [review]
Please look into help text of command `autoscore_ondemand_course_ora`. Rest LGTM 👍

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Review by zeshanarifios — APPROVED** [review]
LGTM 👍

File: common/djangoapps/philu_commands/management/commands/autoscore_ondemand_course_ora.py [hunk 3]
```diff
@@ -1,65 +1,40 @@
...
     help = """
-        Auto Score ORA assesment for On Demand Course if learner has completed all the necessary steps i-e 
-        submitted their responses, assess required peers responses and finally own assessment if required, if that 
-        learner ends up in a waiting queue after 2 weeks of response submission this command will auto-score the
-        learner ORA assessment for those users. 
+    Auto Score ORA assessment for On Demand Course if learner has completed all the necessary steps i-e
+    submitted their responses, assess required peers responses and finally own assessment if required, if that
+    learner ends up in a waiting queue after certain (configurable) days of response submission this command will
+    auto-score the learner ORA assessment for those users. The no of days to wait to auto-score ORA is configurable
+    from site configurations model though its default value is 3 days.
```
[the remark this answers is not in the document]
Comment (zahid-arbisoft) [reply common/djangoapps/philu_commands/management/commands/autoscore_ondemand_course_ora.py]: updated

**Pull request merged**